packages/deployer/ens.js

```diff
@@ -118,6 +185,35 @@ class ENS {
     }
   }
 
+  //this method assumes that the current owner is `from`,
+  //and sets the new owner to be `owner`.
+  async alienateNameOwner({ name, from, owner }) {
+    //setNameOwner goes down the tree, from top to bottom.
+    //here, however, we have to go *up* the tree.
+    //as such, we are going to process the labels in order, rather
+    //than in reverse order.
+
```

Comment:
maybe you said in the description that it is not necessary, but for completeness I'm going to ask, "why do we need to swap owners all the way up the tree?" I mean it's not exactly arbitrary who owns all those names? Every time you use this method all of the names that are subsets of the name you are concerned with will change owners, no?

Reply:
Yeah, I'm really not sure we do!  This is one of the ways that I'm worried that I overcomplicated this!

I mean I guess I just did it this way to keep things parallel and avoid having to write too much new code that doesn't parallel the old code?  But probably this is overcomplicated and should be changed.  I guess I'll go do that?